### ISSUE TITLE
Feature request - Please develop pictures sorting by 'date taken' in gallery app

### ISSUE THREAD

**Issue by happysysadm** [owncloud/apps#503]
Hi,

one of the feature I appreciate more is the possibility to share pictures with friends.

Nonetheless I reckon that this app misses an important feature: in preview mode, the photos are sorted by filename and can't be sorted by date taken, which would be what I expect.

Is it possible to add this feature? 

Thanks in advance!

**Comment by shark0der** [comment]
I installed the app on my phone and was using for a while, so there were let's say 100 pictures with `IMG_20150214_205203` naming scheme - everything was good. Then I decided to upload some pictures that I had on my computer from my old phone. I was used to flash various ROMs on my android and I ended up having to pictures folders: `100MEDIA` and `Camera` with the following naming schemes:

```
$ ls -l 100MEDIA | head -n 5
total 3729916
-rw-rw-r-- 1 anatol anatol  1290309 Feb  9 01:13 IMAG0001.jpg
-rw-rw-r-- 1 anatol anatol   186273 Feb  9 01:14 IMAG0002.jpg
-rw-rw-r-- 1 anatol anatol  1127734 Feb  9 01:14 IMAG0003.jpg
-rw-rw-r-- 1 anatol anatol  1188428 Feb  9 01:14 IMAG0004.jpg
$ ls -l Camera | head -n 5  
total 886668
-rw-rw-r-- 1 anatol anatol 1212437 Feb  9 00:17 1385235567063.jpg
-rw-rw-r-- 1 anatol anatol 1565622 Feb  9 00:17 1394667971351.jpg
-rw-rw-r-- 1 anatol anatol 3841808 Feb  9 00:17 1394791282644.jpg
-rw-rw-r-- 1 anatol anatol 1125071 Feb  9 00:16 1396960177520.jpg
```

In order to upload them, I mounted a folder using webdav and proceed to copy them using
`cp 100MEDIA/* Camera/* /storage/`.

Right now there's a mess. A total mess. When I open the folder I see images from Camera folder, modified 5 hours ago. If I order them backwards, I see the pictures from my new phone, taken about a month ago and I have to scroll through about 100 (thanks God only 100) pictures in order to see the pictures I took yesterday.

The bottom line - alphabetical sort would be a disaster. Ordering by date added to owncloud is ok-ish, but may end-up ugly if people share photos between them (let's say a friend gave me the pictures she took a week ago while at a party). Google Photos is counting massively on the Exif date taken field and I would also expect owncloud to do this. If the field is missing - a fallback to the date when the pictures were added to owncloud would be just good enough.

**Comment by jancborchardt** [comment]
@shark0der »Google Photos is counting massively on the Exif date taken field« – yes, we should also do that.

cc @oparoz @icewind1991

**Comment by oparoz** [comment]
@jancborchardt - See https://github.com/interfasys/galleryplus/issues/91 regarding sorting in general. There are 2 issues:
* Missing GUI
* Missing database component to store meta data

Also, regarding having a iOS type interface, this is discussed here https://github.com/interfasys/galleryplus/issues/95 and not easily doable since the type of app is totally different.

**Comment by DeepDiver1975** [comment]
Please reopen in gallery repo is still valid - THX

**Comment by jancborchardt** [comment]
This is in place in Gallery+ although it could get some improvements by using EXIF dates.
